src/blob/index/core.rs

```diff
@@ -296,27 +290,41 @@ where
     }
 
     async fn get_all(&self, key: &K) -> Result<Option<Vec<RecordHeader>>> {
-        match &self.inner {
+        let res = match &self.inner {
             State::InMemory(headers) => Ok(headers.get(key).cloned()),
             State::OnDisk(findex) => findex.find_by_key(key).await,
-        }
+        };
+        res.map(|r| {
+            r.and_then(|mut hs| {
+                let last_del = (hs.len() - hs.iter().rev().take_while(|h| !h.is_deleted()).count())
+                    .saturating_sub(1);
+                if last_del > 0 {
+                    hs.drain(0..=last_del);
```

Comment:
With the order from newest to latest you need to remove the tail of the vector. That can be done with `truncate` function